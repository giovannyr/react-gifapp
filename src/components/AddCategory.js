import React, { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputValue = (evt) => {
        setInputValue(evt.target.value);
    };

    const handleSubmit = (evt) => {
        evt.preventDefault();
        if(inputValue.trim().length > 2){
            setCategories( categories => [inputValue, ...categories]);
            setInputValue('');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input type="text" value={inputValue} onChange={handleInputValue} />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
