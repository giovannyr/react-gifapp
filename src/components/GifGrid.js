import React from "react";
import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__flipInX animate__slow">
                {category}
            </h3>

            {loading && (
                <p className="animate__animated animate__flash animate__delay-2s">
                    cargando...
                </p>
            )}

            <div className="card-grid">
                {images.map((image) => (
                    <GifGridItem key={image.id} {...image} />
                ))}
            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
};
