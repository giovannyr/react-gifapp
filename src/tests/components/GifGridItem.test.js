import "@testing-library/jest-dom";
import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Pruebas componente GifGridItem", () => {
    const title = "Titulo para pruebas";
    const url = "https://localhost/img.jpg";
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test("debe mostrar <GifGridItem /> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("debe tener un parrafo con el titulo", () => {
        const p = wrapper.find("p");
        expect(p.text().trim()).toBe(title);
    });

    test("debe tener la imagen igual al url y alt de los props", () => {
        const img = wrapper.find("img");
        // console.log(img.props());
        // console.log(img.prop("alt"));
        expect(img.prop("src")).toBe(url);
        expect(img.prop("alt")).toBe(title);
    });

    test("debe tener clase animate__zoomInDown", () => {
        const div = wrapper.find("div");
        const className = div.prop("className");
        expect(className.includes("animate__zoomInDown")).toBe(true);
        // NEGACION
        // expect(className.includes("animate__zoomInDown")).not.toBe(true);
    });
});
