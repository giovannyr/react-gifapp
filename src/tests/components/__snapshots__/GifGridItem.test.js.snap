// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente GifGridItem debe mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__zoomInDown"
>
  <img
    alt="Titulo para pruebas"
    src="https://localhost/img.jpg"
  />
  <p>
    Titulo para pruebas
  </p>
</div>
`;
